src/core/Do.Architecture/Architecture/ApplicationContext.cs

```diff
@@ -5,6 +5,30 @@ public class ApplicationContext
     readonly Dictionary<Type, object> _context = new();
 
     public void Add<T>(T item) where T : notnull => _context[typeof(T)] = item;
-    public T Get<T>() => (T)_context[typeof(T)];
+
+    public T Get<T>()
+    {
+        if (!Has<T>()) throw new KeyNotFoundException(MessageBuilder<T>());
+
+        return (T)_context[typeof(T)];
+    }
+
     public bool Has<T>() => _context.ContainsKey(typeof(T));
+
+    private string MessageBuilder<T>()
```

Comment:
this is like a class name, make it a method name `BuildMessage` or better, `BuildKeyNotFoundMessage`, or even better make this method return exception instance instead of string and name it as `BuildKeyNotFound`